### ISSUE TITLE
Generator loss doubts

### ISSUE THREAD

**Issue by niladell** [coree/ChiNet#26]
I was just going over the generator loss function 
> generator_loss = tf.reduce_sum(-tf.log(score_generated) - cosine_similarity(target_state, generated_state))

 and there are a couple of things I don't fully get.

On the one side the cosine similarity part I assume is a way to make the generated sentence more similar to the objective sentence. Should we really be doing that? On the other side, we don't have any "explicit feedback" from the discriminator, maybe it's late and I just mixing up stuff but, shouldn't the discriminator play an active (or at least explicit) role on the generator loss?

**Comment by axnedergaard** [comment]
The discriminator plays an active role through computation of 'score'. The similarity term in the loss of one of the "GAN hacks" they employ and justify in the paper.
